tests/functional/src/main/java/org/mule/functional/junit4/runners/ArtifactClassLoaderRunner.java

```diff
@@ -82,7 +89,10 @@
      */
     public ArtifactClassLoaderRunner(Class<?> clazz, RunnerBuilder builder) throws Throwable
     {
-        classLoaderTestRunner = createClassLoaderTestRunner(clazz);
+        if (classLoaderTestRunner == null)
+        {
+            classLoaderTestRunner = createClassLoaderTestRunner(clazz);
```

Comment:
This is not correct, classLoaderTestRunner is static but it is created from a particular class. Is not correct to have tests in the same module pointing to different extensions though. This could be fixed if extensions are auto detected. Can you create a jira to to that and reference it here?